examples/darwin-framework-tool/commands/provider/OTAProviderDelegate.mm

```diff
@@ -0,0 +1,173 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "OTAProviderDelegate.h"
+#import <Matter/Matter.h>
+
+constexpr uint8_t kUpdateTokenLen = 32;
+
+@interface OTAProviderDelegate ()
+@property NSString * mOTAFilePath;
+@property DeviceSoftwareVersionModel * candidate;
+@end
+
+@implementation OTAProviderDelegate
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        _nodeID = @(0);
+        _selectedCandidate = [[DeviceSoftwareVersionModel alloc] init];
+        _userConsentState = OTAProviderUserUnknown;
+    }
+    return self;
+}
+
+- (void)handleQueryImage:(MTROtaSoftwareUpdateProviderClusterQueryImageParams * _Nonnull)params
+       completionHandler:(void (^_Nonnull)(MTROtaSoftwareUpdateProviderClusterQueryImageResponseParams * _Nullable data,
+                             NSError * _Nullable error))completionHandler
+{
+    NSError * error;
+    _selectedCandidate.status = @(MTROtaSoftwareUpdateProviderOTAQueryStatusNotAvailable);
+
+    if ([self SelectOTACandidate:params.vendorId rPID:params.productId rSV:params.softwareVersion]) {
+        _selectedCandidate.status = @(MTROtaSoftwareUpdateProviderOTAQueryStatusUpdateAvailable);
```

Comment:
How much test coverage do we want to support here? For instance, I don't see us ever setting `delayedActionTime`. Will we need to support test cases that would result in various values for that field?

Reply:
Not sure if there would be much testing in that since it seems more like a server side test that instructs the server to delay action. If you would like, we can add it to make sure your API is doing the correct thing. The premise of this tool is to validate any code that goes into Matter.framework and not necessarily meet the testing requirement to validate the server. Let me know how you would like to proceed.

Reply:
I created an [issue](https://github.com/project-chip/connectedhomeip/issues/20873) as a follow up.